benchmarks/report.py

```diff
@@ -0,0 +1,52 @@
+import csv
+from abc import ABC, abstractclassmethod
+from dataclasses import dataclass, fields
+from statistics import mean
+from typing import Dict, list, tuple
+
+import numpy as np
+
+duration = int
+
+
+@dataclass
+class MetricCache:
+    epoch_durations: list[duration]
+    batch_durations: list[duration]
+    total_duration: int = 0
+
+
+class MetricExporter(ABC):
```

Comment:
Following up on our discussion from Monday, I took a look at this and I think it should be reasonably simple to plug it into the other `MetricLogger` from the torchvision recipe (it's "metric calculator" more than a "logger", and we can remove the "log" part in favour of this one)